packages/next/next-server/server/next-server.ts

```diff
@@ -992,22 +992,42 @@ export default class Server {
       return { html, pageData, sprRevalidate }
     })
 
-    // render fallback if for a preview path or a non-seeded dynamic path
+    const isProduction = !this.renderOpts.dev
     const isDynamicPathname = isDynamicRoute(pathname)
+    const didRespond = isResSent(res)
+    // const isForcedBlocking =
+    //   req.headers['X-Prerender-Bypass-Mode'] !== 'Blocking'
+
+    // When we did not respond from cache, we need to choose to block on
+    // rendering or return a skeleton.
+    //
+    // * Data requests always block.
+    //
+    // * Preview mode toggles all pages to be resolved in a blocking manner.
+    //
+    // * Non-dynamic pages should block (though this is an be an impossible
+    //   case in production).
+    //
+    // * Dynamic pages should return their skeleton, then finish the data
+    //   request on the client-side.
+    //
     if (
-      !isResSent(res) &&
+      !didRespond &&
       !isDataReq &&
-      ((isPreviewMode &&
-        // A header can opt into the blocking behavior.
-        req.headers['X-Prerender-Bypass-Mode'] !== 'Blocking') ||
-        isDynamicPathname)
+      !isPreviewMode &&
+      isDynamicPathname &&
+      // TODO: development should trigger fallback when the path is not in
+      // `getStaticPaths`, for now, let's assume it is.
+      isProduction
```

Comment:
Are we planning to add this in this PR? If not we should probably leave it as always rendering the fallback in dev mode for dynamic pages until we add this

Reply:
I was not planning on adding that behavior in this PR, as it's a significant divergence from current behavior.

This PR focuses on where we want to be and reverting the client-side loading state behavior.